netplugin/agent/state_event.go

```diff
@@ -289,10 +308,13 @@ func processGlobalConfigUpdEvent(netPlugin *plugin.NetPlugin, opts core.Instance
 		},
 		Instance: opts,
 	}
-	pluginConfig.Instance.ArpMode = cfg.ArpMode
+	pluginConfig.Instance.ArpMode = arpMode
+	if pluginConfig.Instance.FwdMode == "routing" && arpMode == "flood" {
+		log.Infof("Global ARP mode config is not effective when forwarding mode is routing. Proxy-arp will be retained.")
```

Comment:
Moving this to printf will not help as this is in the netplugin logs. I think we can consider printing a similar message when the object is updated.